monai/data/image_reader.py

```diff
@@ -348,3 +349,94 @@ def _get_array_data(self, img: Nifti1Image) -> np.ndarray:
 
         """
         return np.asarray(img.dataobj)
+
+
+class NumpyReader(ImageReader):
+    """
+    Load NPY or NPZ format data based on Numpy library, they can be arrays or pickled objects.
+    A typical usage is to load the `mask` data for classification task.
+    It can load part of the npz file with specified `npz_keys`.
+
+    Args:
+        npz_keys: if loading npz file, only load the specified keys, if None, load all the items.
+            stack the loaded items together to construct a new first dimension.
+
+    """
+
+    def __init__(self, npz_keys: Optional[KeysCollection] = None):
+        super().__init__()
+        self._img: Optional[Sequence[Nifti1Image]] = None
+        if npz_keys is not None:
+            npz_keys = ensure_tuple(npz_keys)
+        self.npz_keys = npz_keys
+
+    def verify_suffix(self, filename: Union[Sequence[str], str]) -> bool:
+        """
+        Verify whether the specified file or files format is supported by Numpy reader.
+
+        Args:
+            filename: file name or a list of file names to read.
+                if a list of files, verify all the subffixes.
+
+        """
+        suffixes: Sequence[str] = ["npz", "npy"]
+        return is_supported_format(filename, suffixes)
+
+    def read(self, data: Union[Sequence[str], str, np.ndarray], **kwargs):
+        """
+        Read image data from specified file or files, or set a Numpy array.
+        Note that the returned object is Numpy array or list of Numpy arrays.
+        `self._img` is always a list, even only has 1 image.
+
+        Args:
+            data: file name or a list of file names to read.
+            kwargs: additional args for `numpy.load` API except `allow_pickle`. more details about available args:
+                https://numpy.org/doc/stable/reference/generated/numpy.load.html
+
+        """
+        self._img = list()
+        if isinstance(data, np.ndarray):
+            self._img.append(data)
+            return data
+
+        filenames: Sequence[str] = ensure_tuple(data)
+        for name in filenames:
+            img = np.load(name, allow_pickle=True, **kwargs)
+            if name.endswith(".npz"):
+                # load expected items from NPZ file
+                npz_keys = [f"arr_{i}" for i in range(len(img))] if self.npz_keys is None else self.npz_keys
```

Comment:
Is there a reason for restricting keys to "arr_{i}" names  if not given? I would use `npz_keys = list(self.npz_keys or img.keys())`,

Reply:
Hi @ericspod ,

"arr_{i}" is the default key name of NPZ data, and we want to keep the order and stack the data, so didn't use `img.keys()`. It's the same behavior with our existing `LoadNumpy` transform. Make sense?

Thanks.

Reply:
The order part of that makes sense. I think `img.keys()` will give you names in order but perhaps best not to rely on it.

Reply:
Yes, I agree.
Thanks.